crates/solidity/outputs/cargo/slang_solidity/src/generated/bindings/generated/binding_rules.rs

```diff
@@ -740,6 +918,76 @@ attribute symbol_reference = symbol  => type = "push_symbol", symbol = symbol, i
 }
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Event definitions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+@event [EventDefinition @name name: [Identifier]] {
+  node def
+  attr (def) node_definition = @name
+  attr (def) definiens_node = @event
+
+  edge @event.def -> def
+}
+
+;; Make the event available to the enclosing contract/interface/library.
+;; NB. top-level events (ie. those defined at the file's level) are already
+;; covered above
+
+@contract [ContractDefinition members: [ContractMembers
+    item: [ContractMember @event variant: [EventDefinition]]
+]] {
+  edge @contract.type_members -> @event.def
+}
+
+@interface [InterfaceDefinition members: [InterfaceMembers
+    item: [ContractMember @event variant: [EventDefinition]]
+]] {
+  edge @interface.type_members -> @event.def
+}
+
+@library [LibraryDefinition members: [LibraryMembers
+    item: [ContractMember @event variant: [EventDefinition]]
+]] {
+  edge @library.members -> @event.def
+}
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Error definitions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+@error [ErrorDefinition @name name: [Identifier]] {
+  node def
+  attr (def) node_definition = @name
+  attr (def) definiens_node = @error
+
+  edge @error.def -> def
+}
+
+;; Make the error available to the enclosing contract/interface/library.
+;; NB. top-level errors (ie. those defined at the file's level) are already
+;; covered above
+
+@contract [ContractDefinition members: [ContractMembers
+    item: [ContractMember @error variant: [ErrorDefinition]]
+]] {
+  edge @contract.type_members -> @error.def
+}
+
+@interface [InterfaceDefinition members: [InterfaceMembers
+    item: [ContractMember @error variant: [ErrorDefinition]]
+]] {
+  edge @interface.type_members -> @error.def
+}
+
+@library [LibraryDefinition members: [LibraryMembers
+    item: [ContractMember @error variant: [ErrorDefinition]]
+]] {
+  edge @library.members -> @error.def
+}
+
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Expressions
```

Comment:
I understand putting the relations between `Expression` and its children/variants here in a dedicated `;;; Expressions` section .. But for relations between parent statements (if/try/for/etc...) and their top-level expression, what do you think about putting it above with the respective statement? I was surprised not seeing them above. Do you think that would make the code more readable?

Reply:
This is an artifact of wanting to preserve strict evaluation available. Because `Expression` nodes are not yet defined when defining the statements, this reorder is not possible with strict evaluation. Since we're not using strict evaluation anymore, it doesn't matter. I'll try to reorganize the rules file to increase readability.